### PR TITLE
Are you sure modal

### PR DESCRIPTION
**Pull request by WarrenK-design**
PR for the Are You Sure Modal when you click "Confirm Trade". 

image

# Buying/Selling Stocks
You can use this at the moment to buy/sell stocks. 

The errors shown will be the ones received from the backend if something is wrong with the request, there is no checks on frontend yet. 

image
image

# Known Issues in Other Areas 
Submitted this as a PR before went fixing issues in other areas of trade page so you may notice some of these things when clicking through, let me know if see anymore. 

## Buy Sell Buttons
When you go onto page first nothing visually shown that default buy and market order is selected but it is. 
image
Need to break out Buy/Sell into own component too as discussed in Discord. 
image

## Quantity 
This defaults to ```NaN``` wen you first go onto the page should default to 0 or 1. 
image
Maybe change the styling of it to the game sliders? Not sure if be too tricky or not. 
image
image

## Portfolio Balance 
When you select ```Sell``` the viz updates as if you are buying, should increase the "Left To Spend" balance as I am selling stocks not buying stocks might be tricky one. 
image

## Numbers Not Adding Up 
For the "Are You Sure" component these numbers will not make sense for few reasons. The sell orders issue listed above looks like you losing money. 
image

On the frontend the fees aren't being added in either for the trade yet, in the success message this is the up to date portfolio amount from the backend but the frontend wont match for this reason. 
image

### CODE REVIEW

File: src/components/confirmOrderComponents/OrderSummary.js [hunk 1]
```diff
@@ -18,20 +21,19 @@ function OrderSummary({ buyOrSell, orderType, amountSelected, newPortfolioBalanc
...
                             </tr>
                             <tr>
                                 <td>New Portfolio Balance</td>
-                                <td>${parseFloat(newPortfolioBalance).toFixed(2)}</td>
+                                <td>${parseFloat(newPortfolioBalance).toLocaleString('en-US', {minimumFractionDigits: 2, maximumFractionDigits: 2})}</td>
                             </tr>
                             <tr>
                                 <td>Total Cost</td>
-                                <td>${parseFloat(amountSelected).toFixed(2).toString()}</td>
+                                <td>${parseFloat(amountSelected).toLocaleString('en-US', {minimumFractionDigits: 2, maximumFractionDigits: 2})}</td>
```
Comment (joeycor92) [inline src/components/confirmOrderComponents/OrderSummary.js:32]: Just wondering why this is EN-US and not EN-IR or whatever the Irish version is?

**Review by joeycor92 — APPROVED** [review]
Happy with this. As you mentioned, loads of work to do, but at least we can buy and sell now, which is just massive. 

Thanks for fixing the backend errors as well, they make a lot more sense to a user now.

File: src/components/confirmOrderComponents/OrderSummary.js (hunk 1 above)
Comment (WarrenK-design) [reply src/components/confirmOrderComponents/OrderSummary.js:32]: Just went with the US one because everything is in dollars in the game.

**Comment by beckskelly** [comment]
Hey Warren, great work

I went to buy a second stock, my available balance was showing incorrectly, is there any way of sorting that within this PR?

image

**Comment by beckskelly** [comment]
hey @WarrenK-design , after discussion in thread, I can see this was mostly a functionality pr, so I'll approve this

Another issue on that trade page is just adding infobuttons everywhere, like what is a market order, limit order etc

Cheers for all the work

**Review by beckskelly — APPROVED** [review]

**Pull request merged**